File: src/main/java/com/leon/bugreport/expansions/BugPlaceholders.java
package com.leon.bugreport.expansions;

import com.leon.bugreport.API.ErrorClass;
import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import org.bukkit.OfflinePlayer;
import org.bukkit.plugin.Plugin;
import org.jetbrains.annotations.NotNull;

import static com.leon.bugreport.BugReportDatabase.*;
import static com.leon.bugreport.BugReportManager.debugMode;

public class BugPlaceholders extends PlaceholderExpansion {
	private final Plugin plugin;

	public BugPlaceholders(Plugin plugin) {
		this.plugin = plugin;
	}

	@Override
	public boolean persist() {
		return true;
	}

	@Override
	public boolean canRegister() {
		return true;
	}

	@Override
	public String getAuthor() {
		return plugin.getDescription().getAuthors().toString();
	}

	@Override
	public String getIdentifier() {
		return "bugreports";
	}

	@Override
	public String getVersion() {
		return plugin.getDescription().getVersion();
	}

	@Override
	public String onRequest(OfflinePlayer player, @NotNull String params) {
		if (debugMode) {
			ErrorClass.throwDebug("LinkDiscordCommand: Requested Placeholders", "debug");
		}
		switch (params) {
			case "totalBugReports" -> {
				return String.valueOf(loadBugReportCount());
			}
			case "totalArchivedBugReports" -> {
				return String.valueOf(loadArchivedBugReportCount());
			}
			case "totalNonArchivedBugReports" -> {
				return String.valueOf(loadNonArchivedBugReportCount());
			}
			case "totalDeletedBugReports" -> {
				return String.valueOf(loadDeletedBugReportCount());
			}
			default -> {
				return null;
			}
		}
	}
}

File: src/main/java/com/bugreportmc/bugreport/expansions/BugPlaceholders.kt
package com.bugreportmc.bugreport.expansions

import com.bugreportmc.bugreport.BugReportDatabase.Companion.loadArchivedBugReportCount
import com.bugreportmc.bugreport.BugReportDatabase.Companion.loadBugReportCount
import com.bugreportmc.bugreport.BugReportDatabase.Companion.loadDeletedBugReportCount
import com.bugreportmc.bugreport.BugReportDatabase.Companion.loadNonArchivedBugReportCount
import me.clip.placeholderapi.expansion.PlaceholderExpansion
import org.bukkit.OfflinePlayer
import org.bukkit.plugin.Plugin

class BugPlaceholders(plugin: Plugin) : PlaceholderExpansion() {
	override fun persist(): Boolean {
		return true
	}

	override fun canRegister(): Boolean {
		return true
	}

	private val author: String = plugin.description.authors.first()
	private val version: String = plugin.description.version
	private val identifier: String = "bugreports"

	override fun onRequest(player: OfflinePlayer?, params: String): String? {
		return when (params) {
			"totalBugReports" -> {
				loadBugReportCount().toString()
			}

			"totalArchivedBugReports" -> {
				loadArchivedBugReportCount().toString()
			}

			"totalNonArchivedBugReports" -> {
				loadNonArchivedBugReportCount().toString()
			}

			"totalDeletedBugReports" -> {
				loadDeletedBugReportCount().toString()
			}

			else -> {
				null
			}
		}
	}

	/**
	 * The placeholder identifier of this expansion. May not contain %,
	 * {} or _
	 *
	 * @return placeholder identifier that is associated with this expansion
	 */
	override fun getIdentifier(): String {
		return identifier
	}

	/**
	 * The author of this expansion
	 *
	 * @return name of the author for this expansion
	 */
	override fun getAuthor(): String {
		return author
	}

	/**
	 * The version of this expansion
	 *
	 * @return current version of this expansion
	 */
	override fun getVersion(): String {
		return version
	}
}
